### PR TITLE
check if values are true not just undefined

### LINKED ISSUE #155336: Task command not registered on load
1. Open a folder
2. Immediately try to run task
3. 🐛 Says task command not found

Recording 2022-07-15 at 14 48 02

### PR DESCRIPTION
**Pull request by meganrogge**
part of #155227
fixes #155336

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 1]
```diff
@@ -324,6 +326,7 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	public registerSupportedExecutions(custom?: boolean, shell?: boolean, process?: boolean) {
+		TaskCommandsRegistered.bindTo(this._contextKeyService);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:329]:
```suggestion
```

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 2]
```diff
@@ -491,7 +493,7 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 				this._openTaskFile(resource, TaskSourceKind.WorkspaceFile);
 			}
 		});
-		TaskCommandsRegistered.bindTo(this._contextKeyService).set(true);
+		console.log('registered commands', TaskCommandsRegistered.getValue(this._contextKeyService));
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:496]:
```suggestion
TaskCommandsRegistered.getValue(this._contextKeyService));
```

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 3]
```diff
@@ -491,7 +493,7 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 				this._openTaskFile(resource, TaskSourceKind.WorkspaceFile);
 			}
 		});
-		TaskCommandsRegistered.bindTo(this._contextKeyService).set(true);
+TaskCommandsRegistered.getValue(this._contextKeyService));
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:496]:
```suggestion
```

**Review by roblourens — APPROVED** [review]

**Comment by joaomoreno** [comment]
FYI reverted this due to build breakage: https://github.com/microsoft/vscode/pull/155468

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 4]
```diff
@@ -2173,7 +2174,7 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	private get _jsonTasksSupported(): boolean {
-		return !!ShellExecutionSupportedContext.getValue(this._contextKeyService) && !!ProcessExecutionSupportedContext.getValue(this._contextKeyService);
+		return ShellExecutionSupportedContext.getValue(this._contextKeyService) === true && ProcessExecutionSupportedContext.getValue(this._contextKeyService) === true && !Platform.isWeb;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:2177]: This `!Platform.isWeb` was the problem:

```
  5 failing

  1) VSCode Smoke Tests (Web)
       Task
         Task Quick Pick
           Tasks: Run Task
             hide property - false:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at Code.poll (D:\a\_work\1\s\test\automation\src\code.ts:255:11)
      at Code.waitForElements (D:\a\_work\1\s\test\automation\src\code.ts:204:10)
      at QuickInput.waitForQuickInputElements (D:\a\_work\1\s\test\automation\src\quickinput.ts:41:3)
      at Task.assertTasks (D:\a\_work\1\s\test\automation\src\task.ts:42:4)
      at Context.<anonymous> (src\areas\task\task-quick-pick.test.ts:36:5)

  2) VSCode Smoke Tests (Web)
       Task
         Task Quick Pick
           Tasks: Run Task
             hide property - undefined:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at Code.poll (D:\a\_work\1\s\test\automation\src\code.ts:255:11)
      at Code.waitForElements (D:\a\_work\1\s\test\automation\src\code.ts:204:10)
      at QuickInput.waitForQuickInputElements (D:\a\_work\1\s\test\automation\src\quickinput.ts:41:3)
      at Task.assertTasks (D:\a\_work\1\s\test\automation\src\task.ts:42:4)
      at Context.<anonymous> (src\areas\task\task-quick-pick.test.ts:40:5)

  3) VSCode Smoke Tests (Web)
       Task
         Task Quick Pick
           Tasks: Run Task
             icon - icon only:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at Code.poll (D:\a\_work\1\s\test\automation\src\code.ts:255:11)
      at Code.waitForElements (D:\a\_work\1\s\test\automation\src\code.ts:204:10)
      at QuickInput.waitForQuickInputElements (D:\a\_work\1\s\test\automation\src\quickinput.ts:41:3)
      at Task.assertTasks (D:\a\_work\1\s\test\automation\src\task.ts:42:4)
      at Context.<anonymous> (src\areas\task\task-quick-pick.test.ts:45:5)
```

File: src/vs/workbench/contrib/tasks/browser/task.contribution.ts [hunk 5]
```diff
@@ -40,7 +40,7 @@ import { TaskDefinitionRegistry } from 'vs/workbench/contrib/tasks/common/taskDe
 import { TerminalMenuBarGroup } from 'vs/workbench/contrib/terminal/browser/terminalMenus';
 import { isString } from 'vs/base/common/types';
 
-const SHOW_TASKS_COMMANDS_CONTEXT = ContextKeyExpr.or(ShellExecutionSupportedContext, ProcessExecutionSupportedContext);
+const SHOW_TASKS_COMMANDS_CONTEXT = ContextKeyExpr.and(ShellExecutionSupportedContext, ProcessExecutionSupportedContext, TaskCommandsRegistered);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/tasks/browser/task.contribution.ts:43]: I'm pretty sure `or` is correct here

**Comment by Tyriar** [comment]
Second try: https://github.com/microsoft/vscode/pull/155524

**Pull request merged**